src/main/webapp/definitions/alfresco-core.yaml

```diff
@@ -11001,6 +11113,64 @@ definitions:
         type: array
         items:
           $ref: '#/definitions/ActionBodyExecTemplate'
+  RuleSetPaging:
+    type: object
+    required:
+      - list
+    properties:
+      list:
+        type: object
+        required:
+          - pagination
+          - entries
+        properties:
+          pagination:
+            $ref: '#/definitions/Pagination'
+          entries:
+            type: array
+            items:
+              $ref: '#/definitions/RuleSetEntry'
+  RuleSetEntry:
+    type: object
+    required:
+      - entry
+    properties:
+      entry:
+        $ref: '#/definitions/RuleSet'
+  RuleSet:
+    type: object
+    required:
+      - id
+      - ownedBy
+    properties:
+      id:
+        description: Identifier for the rule set
+        type: string
+      ownedBy:
+        description: The identifier of the folder which owns the rule set
+        type: string
+      sharedWith:
+        description: List of folder identifiers which also use the rule set
```

Comment:
I assume this is restricted to just the ids of folders that the user has access to (but I don't know for sure - e.g. we should check how some of the other endpoints behave).

Reply:
Yes, good point @tpage-alfresco - in theory, we only return folder ids that the user has read permission. 

Implies that "ownedBy" may not be returned in some cases. 

Similarly it would imply "sharedWith" may have less entries (even zero). However, makes it difficult to know if there are any hidden links ?

Reply:
>  one use case for this would be to show a warning in case we're modifying a rule set that may be used by other folders

Yes, exactly. Or do we mask with a dummy/fake id in the list "****" ? 

We could also check what V0 does in this situation (although may not be desired or solve our proposed use case).